hack/setup-kind-in-ci.sh

```diff
@@ -0,0 +1,157 @@
+-iv|--istio-version <#.#.#>
+    The version of Istio you want to install.
+    This option is ignored if -ii is false.
+    If not specified, the latest version of Istio is installed.
+    Default: <the latest release>
+HELP
+}
+
+# process command line arguments
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -dorp|--docker-or-podman)     DORP="$2";                  shift;shift; ;;
+    -h|--help)                    helpmsg;                    exit 1       ;;
+    -iv|--istio-version)          ISTIO_VERSION="$2";         shift;shift; ;;
+    *) echo "Unknown argument: [$key]. Aborting."; helpmsg; exit 1 ;;
+  esac
+done
+
+# abort on any error
+set -e
+
+# set up some of our defaults
+DORP="${DORP:-docker}"
+
+# Only if this is set to "true" will Istio be installed if it is missing
+INSTALL_ISTIO="${INSTALL_ISTIO:-true}"
+
+# print out our settings for debug purposes
+cat <<EOM
+=== SETTINGS ===
+DORP=$DORP
+ISTIO_VERSION=$ISTIO_VERSION
+=== SETTINGS ===
+EOM
+
+infomsg "Make sure everything exists"
+which kubectl > /dev/null || (infomsg "kubectl executable is missing"; exit 1)
+which kind > /dev/null || (infomsg "kind executable is missing"; exit 1)
+which "${DORP}" > /dev/null || (infomsg "[$DORP] is not in the PATH"; exit 1)
+
+infomsg "Kind cluster to be created with name [ci]"
+cat <<EOF | kind create cluster --name ci --config -
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+  - role: worker
+EOF
+
+# When a new cluster is created, kind automagically sets the kube context.
+
+infomsg "Create Kind LoadBalancer via MetalLB"
+lb_addr_range="255.70-255.84"
+
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.12.1/manifests/namespace.yaml
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.12.1/manifests/metallb.yaml
+
+subnet=$(${DORP} network inspect kind --format '{{(index .IPAM.Config 0).Subnet}}')
+subnet_trimmed=$(echo "${subnet}" | sed -E 's/([0-9]+\.[0-9]+)\.[0-9]+\..*/\1/')
+first_ip="${subnet_trimmed}.$(echo "${lb_addr_range}" | cut -d '-' -f 1)"
+last_ip="${subnet_trimmed}.$(echo "${lb_addr_range}" | cut -d '-' -f 2)"
+cat <<LBCONFIGMAP | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    address-pools:
+    - name: default
+      protocol: layer2
+      addresses: ['${first_ip}-${last_ip}']
+LBCONFIGMAP
+
+# if [ -n "${ISTIO_VERSION}" ]; then
+# DOWNLOAD_ISTIO_VERSION_ARG="--istio-version ${ISTIO_VERSION}"
+# fi
+
+infomsg "Downloading istio"
+hack/istio/download-istio.sh
+
+infomsg "Installing istio"
+hack/istio/install-istio-via-istioctl.sh --client-exe-path "$(which kubectl)"
+  
+infomsg "Pushing the images into the cluster..."
+make -e DORP="${DORP}" -e CLUSTER_TYPE="kind" -e KIND_NAME="ci" cluster-push-kiali
+
+infomsg "Cloning kiali helm-charts..."
+git clone --single-branch --branch master https://github.com/kiali/helm-charts.git helm-charts
```

Comment:
does master here assume this was triggered by a change to the master branch? Shouldn't this use the GITHUB_REF_NAME? Note that the GITHUB_REF might reference a PR branch name, so I'm not sure what the right thing to do is.

Reply:
hmmm we probably want the target branch name which would be master if the PR is against the master branch and something like `v1.48` if it's a feature branch. I think the target branch name is an env var too.